python/modules/IcePy/Communicator.cpp

```diff
@@ -70,112 +71,22 @@ communicatorNew(PyTypeObject* type, PyObject* /*args*/, PyObject* /*kwds*/)
     return self;
 }
 
+// Only called by Ice.Initialize
+// IcePy.Communicator(args: [str]|None, initData: Ice.InitializationData|None, configFile: str|None)
+
 extern "C" int
 communicatorInit(CommunicatorObject* self, PyObject* args, PyObject* /*kwds*/)
 {
-    //
-    // The argument options are:
-    //
-    // Ice.initialize()
-    // Ice.initialize(args)
-    // Ice.initialize(initData)
-    // Ice.initialize(configFile)
-    // Ice.initialize(args, initData)
-    // Ice.initialize(args, configFile)
-    //
-
-    PyObject* arg1{nullptr};
-    PyObject* arg2{nullptr};
-    if (!PyArg_ParseTuple(args, "|OO", &arg1, &arg2))
-    {
-        return -1;
-    }
-
     PyObject* argList{nullptr};
     PyObject* initData{nullptr};
     PyObject* configFile{nullptr};
-
-    if (arg1 == Py_None)
-    {
-        arg1 = nullptr;
-    }
-
-    if (arg2 == Py_None)
-    {
-        arg2 = nullptr;
-    }
-
-    PyObject* initDataType = lookupType("Ice.InitializationData");
-
-    if (arg1)
+    if (!PyArg_ParseTuple(args, "OOO", &argList, &initData, &configFile))
```

Comment:
Switched this internal function to have 3 positional arguments, as mentioned in the comment above.

Reply:
Because this is an internal function and the only caller is Ice.initialize, I moved the validation logic there in Python code which is simpler.

I can add an assert here to ensure that we never pass both. Having 3 parameters is simpler because we don't have to do the additional type checking here.

Reply:
I would keep using data here as a union: `data: Union[Ice.InitializationData,str,null]`

Reply:
> I would keep using data here as a type union

This would require to check the types here again, and we already do in Ice.initialize. I don't see the upside of using an union.